src/components/HeaderInfo/HeaderInfo.js

```diff
@@ -215,6 +215,10 @@ export const HeaderInfo = ({
   const [selectedReportingPeriods, setSelectedReportingPeriods] = useState(
     currentTab?.reportingPeriods ?? []
   );
+  /* defaultReportingPeriodChanged is a flage; user selecte a reporing date or not;
+    it's changed when user manually change location or viewTemplate or when import data
+  */
+  const [defaultReportingPeriodChanged, setDefaultReportingPeriodChanged] = useState(false);
```

Comment:
Is this flag really necessary? The issue seems to be resolved without it. Correct me if I'm wrong, but it looks like the main fix is in concatenating the new period with the previously selected ones.

Reply:
When a user goes directly to the imported data, the user has not selected the current year and last quarter from the reporting period list. Therefore, it is necessary to prevent it. That is why I use a flag.

Reply:
Ah okay, I get it now, thanks for clarifying!